### ISSUE TITLE
Tensorflow freezes for a small model in Windows 10

### ISSUE THREAD

**Issue by pengd49** [tensorflow/tensorflow#7320]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
--> I have not been able to find any related information. I did post a SO thread (http://stackoverflow.com/questions/41889147/tensorflow-execution-freezes-for-a-small-cnn), and got the suggestion that I may have encountered a bug.

### Environment info
Operating System: 
--> Windows 10 home edition

Installed version of CUDA and cuDNN: 
--> None. Was using the CPU version.
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):

If installed from binary pip package, provide:

1. A link to the pip package you installed: 
--> https://storage.googleapis.com/tensorflow/windows/cpu/tensorflow-0.12.1-cp35-cp35m-win_amd64.whl
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
--> 0.12.1

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
--> Attached as "Issue_example_code.txt"
A brief explanation of the issue: 
Training freezes as I slightly increase the model size. For the attached code, using 50, 50 for the two fully-connected layers would work, but going slightly larger, e.g., to 100, 50, would result in execution freezing after model initialization or a few rounds of training. It does not seem to be constrained by memory, since when the freeze occurs the Python process takes around 50MB of memory (as observed from task manager), while when the training runs (for smaller sizes) it takes GB range of memory.

### What other attempted solutions have you tried?
--> None, as I was unable to find related info by searching and I myself does not understand the possible cause of this issue.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

[Issue_example_code.txt](https://github.com/tensorflow/tensorflow/files/756669/Issue_example_code.txt)

**Comment by girving** [comment]
Can you get us a stack trace from where it freezes?  This isn't enough information to guess at a cause.  A Python stack trace would be a starting point; just hit ctrl-c or whatever the windows equivalent is.

**Comment by pengd49** [comment]
@girving Actually the ctrl-c doesn't work in this case: the console/execution simply stays frozen. I had to forcibly close the console or kill the process to terminate the execution.

Is there another good way to gather more info?

**Comment by girving** [comment]
The ideal would be to run it through a debugger.  You could also try adding print statements until you localize it, but it'll be hard to get below the Python level with those.

@mrry Do you have suggestions for Windows debugging?

**Comment by mrry** [comment]
I've mainly used Visual Studio for debugging on Windows. The free Community edition should be able to attach to your frozen process and gather a stack trace.

**Comment by pengd49** [comment]
I have stepped through my code in the IDLE debugger and gotten the following stack trace:

'__main__'.<module>(), line 87:_, l1, l2, predictions = session.run([optimizer, loss, loss_reg, train_prediction], feed_dict=feed_dict)
'tensorflow.python.client.session'.run(), line 766: run_metadata_ptr)
'tensorflow.python.client.session'._run(), line 964: feed_dict_string, options, run_metadata)
'tensorflow.python.client.session'._do_run(), line 1014: target_list, options, run_metadata)
'tensorflow.python.client.session'._do_call(), line 1021: return fn(*args)
'tensorflow.python.client.session'._run_fn(), line 1003: status, run_metadata)

The execution then freezes at line 1001 when I tried to step into tf_session.TF_Run(...) there. Should I go deeper?

**Comment by girving** [comment]
@pengd49 Yes, unfortunately `TF_Run` is where everything happens, so it isn't specific enough yet.

**Comment by pengd49** [comment]
@girving If I understand it correctly, the TF_Run is ultimately defined in the file _pywrap_tensorflow.pyd, right? 

I'm sorry I am inexperienced in debugging a pyd file. From what I found online, it seems I will need to try to compile and generate a version of the pyd with debug flag in order to debug it?

**Comment by girving** [comment]
@pengd49 `TF_Run` is where essentially everything happens.  All it means is that the error is somewhere in C++ code.

**Comment by aselle** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
